CMakeLists.txt

```diff
@@ -20,6 +20,29 @@ endif ()
 project(remill)
 cmake_minimum_required(VERSION 3.2)
 
+if(EMSCRIPTEN)
+  set(CMAKE_BUILD_TYPE Release)
+  set(gflags_DIR "$ENV{REPOS}/gflags/build")
+  set(glog_DIR "$ENV{REPOS}/glog/build")
+  set(LLVM_DIR "$ENV{REPOS}/llvm-project/llvm/build/lib/cmake/llvm/")
+  set(remill_DIR "$ENV{REPOS}/remill/build/")
+
+  # We want to manually invoke main for Lift instead of Emscripten calling it for us
+  # So we disable INVOKE_RUN, export callMain (and FS for files).
+  set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/web/src")
+  set(CMAKE_CXX_FLAGS "\
```

Comment:
Is it possible to move these settings in cmake/settings.cmake and use GLOBAL_CXX_FLAGS instead of CMAKE_CXX_FLAGS?

Reply:
I'll certainly try. I don't see any reference to `GLOBAL_CXX_FLAGS` in CMake docs, or used anywhere in Remill. I also don't see it being respected anywhere in Emscripten. Is this a feature I'm unaware of?

Reply:
The goal is trying to not edit the CMake CXX variables by hand, and append them to the variables used there (which will eventually use property-based target settings such as `target_compile_options`)

I think it is fine to append them directly with something similar to

```cmake
if(EMSCRIPTEN)
  list(GLOBAL_CXX_FLAGS APPEND
    flag1
    flag2
  )
endif()
```
but you can also create a setting such as `REMILL_EMSCRIPTEN_CXXFLAGS` if you think it may be useful